src/shared/lib/ExtendedRouterStore.ts

```diff
@@ -35,16 +32,13 @@ export default class ExtendedRouterStore extends (RouterStore as FunctionConstru
             }
         });
 
+        // note that param handles encoding of parameter values
         let params = $.param(newQuery);
 
-        // if we have a new path, adopt it
-        // otherwise use current path
-
-        newPath = newPath || this.location.pathname;
-
-        newPath = (newPath[0] === '/') ? newPath : '/'+ newPath;
+        // put a leading slash if there isn't one
+        path = (path[0] === '/') ? path : '/' + path;
```

Comment:
this is the kind of stuff that leads to issues in edge cases but this is probably simple enough to be fine